lychee-lib/src/types/cache.rs

```diff
@@ -16,6 +16,11 @@ pub enum CacheStatus {
     /// The request was excluded (skipped)
     Excluded,
     /// The protocol is not yet supported
+    // We no longer cache unsupported files as they might be supported in future
+    // versions.
+    // Nevertheless, keep for compatibility when deserializing older cache
+    // files, even though this no longer gets serialized. Can be removed at a
+    // later point in time.
```

Comment:
This is a tangential discussion but have you all considered versioning the `.lycheecache` file to allow for "breaking" changes in the file format? 

I thought about that while working on the cache deserialization fix but it seemed unwise to break the expected format for a simple bug fix

Reply:
this is a good idea.

Reply:
Agree, it's a solution. How can csv files be versioned? Would we just add a header to the file? In that case, would that still be a valid csv file?

Reply:
@mre 

just add comment lines, for example

#Cache file generated by lychee
#Do NOT modify it by hand
#Timestamp=xxxxxxx
#Version=1

Reply:
the drawback is that CSV standard doesn't specify comment syntax

so some application like Excel see them as actual data.

but decent parsers usually allows you to specify comment character.

Reply:
How would the version metadata be handled by `lychee` when encountering an older version?

If you're not providing backward-compatibility, it would fail either way?

Alternatively you could add a version suffix to the filename itself, or just switch away from CSV? Is there anyone requiring CSV? Presumably `.lycheecache` would only ever be intended to be parsed by `lychee`.

I think we had some simpler approaches to handle the recent serde issues for handling the cache, but ended up with a custom deserializer because of support issues with serde for CSV with enums. If you're going to introduce versioning that would make existing format redundant, might as well consider a different format for `.lycheecache` too.

Reply:
we can have backward-compatibility.
mostly likely the breaking changes are:
- Changing cache format
- Adding/removing keys or re-assigning meaning of keys
- Including/excluding queries in/from cache

---

`.lycheecache` doesn't even't need to human readable. binary format would be smaller and faster to load.

Reply:
The reason why I like CSV is because it's simple.
The manual editability is a nice benefit: users could retry a few URLs by removing lines from the file without invalidating the rest of the cache. I make use of that trick sometimes.

The fact that it's human-readable also means that other Unix tools can work with the data:
```bash
# How many non-200 status codes were cached?
grep -v "200" .lycheecache | wc -l
```

It also compresses really well with GZIP, so file size isn't a major concern. For instance, a cache for the top 10k websites is 358K and 88K gzipped (25% of original size). With brotli it's 72K.

We could change the cache filetype, but I don't know of a better alternative that still retains these properties. 
I'd be okay with adding a version as a comment to CSV &mdash; at least it seems to be a relatively common way to add metadata. Then again I wonder if we should postpone the decision until we hit our first breaking change with the existing format.
In any case, the review section is probably not the right place for making a decision, so maybe someone wants to open up an issue and summarize the discussion? 😅